### PR TITLE
added shell setup code

### PR DESCRIPTION
**Pull request by prune998**
Added some shell setup including zsh and theme

### CODE REVIEW

File: docs/shell_setup.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
+# Shell Setup
+
+Ok, now you have installed `kubectl` and you have two working local clusters. You will soon realize that you're type `kubectl` (7 letters) quite often. Too often.
+
+You may also wonder *which cluster am I connected to ?* or *what is my default namespace ?*.
+
+You can east this pain by configuring your shell.
```
Comment (archyufa) [inline docs/shell_setup.md]: east -> ease

File: docs/shell_setup.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
+# Shell Setup
+
+Ok, now you have installed `kubectl` and you have two working local clusters. You will soon realize that you're type `kubectl` (7 letters) quite often. Too often.
+
+You may also wonder *which cluster am I connected to ?* or *what is my default namespace ?*.
+
+You can east this pain by configuring your shell. 
+
+
+## completion
```
Comment (archyufa) [inline docs/shell_setup.md]: completion -> Completion

File: docs/shell_setup.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    ```
+
+=== "BASH"
+    ```bash linenums="1" title="~/.bashrc"
+    complete -C '/usr/local/bin/aws_completer' aws
+    ```
+
+## ZSH
+
+Mac OSX shell is now `zsh` by default. So we'll focus on ZSH now. ZSH is also widely available in Linux and may somtimes be the default too.
```
Comment (archyufa) [inline docs/shell_setup.md]: Mac OSX shell is now -> Mac OSX shell is now uses

File: docs/shell_setup.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+```bash
+sh -c "$(curl -fsSL https://raw.github.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"
+```
+
+#### Theme
+
+The best theme I found is called `PowerLevel10k`. It's easy to install and provide a ton of tooling in your shell. It's available in a git repo that you have to clone in your `.oh-my-zsh` custom folder.
+
+!!! note "Fonts"
+    If you are using iTerm2 or Termux, p10k configure can install the recommended font for you. Simply answer Yes when asked whether to install Meslo Nerd Font.
```
Comment (archyufa) [inline docs/shell_setup.md]: not sure if  you need  work `configure`  after `p10k`

File: docs/shell_setup.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+Mac OSX shell is now `zsh` by default. So we'll focus on ZSH now. ZSH is also widely available in Linux and may somtimes be the default too.
+
+!!! note "ZSH deprecation warning"
+    To get rid of the deprecation warning in OSX, export this variable in your shell. I recommend adding it to your `.zshrc` file:
+
+    ```bash linenums="1" title="~/.zshrc"
+    export BASH_SILENCE_DEPRECATION_WARNING=1
+    ```
+### Oh My ZSH !
```
Comment (archyufa) [inline docs/shell_setup.md:91]: Can we have a simple description of `Oh My ZSH` ? And we need this.

File: docs/shell_setup.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+#### install
+
+```bash
+sh -c "$(curl -fsSL https://raw.github.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"
+```
+
+#### Theme
+
+The best theme I found is called `PowerLevel10k`. It's easy to install and provide a ton of tooling in your shell. It's available in a git repo that you have to clone in your `.oh-my-zsh` custom folder.
```
Comment (archyufa) [inline docs/shell_setup.md]: i, we, our, us, you, your –> rewrite sentence

There are many great themes out there to help developers to become more productive and look cool.
On such theme is `PowerLevel10k`. It's easy to install and provides a ton of tooling in your shell. It's available in a git repo that you have to clone in your `.oh-my-zsh` custom folder.

File: docs/shell_setup.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+=== "Linux"
+
+    ```bash
+    git clone --depth=1 https://github.com/romkatv/powerlevel10k.git ${ZSH_CUSTOM:-$HOME/.oh-my-zsh/custom}/themes/powerlevel10k
+    sed 's/^ZSH_THEME=".*"/ZSH_THEME="powerlevel10k\/powerlevel10k"/g' -i ~/.zshrc
+    zsh
+    ```
+
+When opening a new ZSH shell for the first time (done by calling `zsh` above), you'll be asked a lot of questions to setup your shell to your convenience.
```
Comment (archyufa) [inline docs/shell_setup.md]: i, we, our, us, you, your –> rewrite sentence

File: docs/shell_setup.md [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+
+You can re-enter this setup phase anytime with the command `p10k configure`
```
Comment (archyufa) [inline docs/shell_setup.md]: i, we, our, us, you, your –> rewrite sentence

File: docs/kubectl_tooling/kubectl.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
+# Kubectl
+
+You may not expect to see Kubectl in the `tooling` section, as it, well, the basic. 
+
+But maybe you're using it wrong ? Or you're not using it to its full potential ? 
+
+We're not going to dive in too much, but here are some cool usage of `kubectl`.
```
Comment (archyufa) [inline docs/kubectl_tooling/kubectl.md]: i, we, our, us, you, your –> rewrite sentence

**Comment by archyufa** [comment]
Some minor suggestion please review.

**Review by archyufa — COMMENTED** [review]
sharing my feedback

File: docs/shell_setup.md (hunk 4 above)
Comment (prune998) [reply docs/shell_setup.md]: this is the comment taken from the official repo :)

File: docs/shell_setup.md (hunk 8 above)
Comment (prune998) [reply docs/shell_setup.md]: do you mean `you` is not OK ?

Comment (prune998) [reply docs/shell_setup.md]: ok, updated

File: docs/shell_setup.md [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+    To get rid of the deprecation warning in OSX, export this variable in your shell. Add it to your `.zshrc` file for persistance:
+
+    ```bash linenums="1" title="~/.zshrc"
+    export BASH_SILENCE_DEPRECATION_WARNING=1
+    ```
+### Oh My ZSH !
+
+Oh My Zsh is a delightful, open source, community-driven framework for managing your Zsh configuration. It comes bundled with thousands of helpful functions, helpers, plugins, themes, and a few things that make you shout... `"Oh My ZSH!"`
+
+#### install
```
Comment (archyufa) [inline docs/shell_setup.md]: install  -> Install `Oh My ZSH` framework

File: docs/shell_setup.md [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+
+```bash
+sh -c "$(curl -fsSL https://raw.github.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"
+```
+
+#### Theme
```
Comment (archyufa) [inline docs/shell_setup.md:101]: Theme -> Setup `PowerLevel10k` Theme for `Oh My ZSH`

File: docs/shell_setup.md [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+- `Prompt Connection`: (if `Prompt Height` is set to 2): Disconnected (1)
+- `Prompt Frame`: (if `Prompt Height` is set to 2): No Frame (1)
+- `Prompt Spacing`: Compact (1)
+- `Icons`: Many (2)
+- `Prompt Flow`: Concise (1)
+- `Enable Transient Prompt?`: Yes (y)
+- `Instant Prompt Mode`: speed up slow scripts... set to Verbose as recommended (1)
+- `Apply changes to ~/.zshrc`: Damn YES (y)
+
+#### Plugins
```
Comment (archyufa) [inline docs/shell_setup.md:155]: Plugins -> Configure `Oh My ZSH` Plugins

File: docs/shell_setup.md [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+
+ZSH uses plugins that extend the shortcuts and other behaviours for default apps. Here's the list I usually use:
+
+```yaml
```
Comment (archyufa) [inline docs/shell_setup.md:159]: Why Vault and terraform here? 

Can we add following instead:
`zsh-autosuggestions zsh-syntax-highlighting kube-ps1 kubectl gcloud`

File: docs/shell_setup.md [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+plugins=(brew kubectl git python tmux vault terraform)
+```
+
+Once those plugins are installed and the shell is reloaded, a ton of new Aliases are created. Use `alias` command to list them all. Here are a few:
```
Comment (archyufa) [inline docs/shell_setup.md:163]: Can we have demo's for zsh-autosuggestions and other plugins ?

**Review by archyufa — COMMENTED** [review]
adding few comments

File: docs/shell_setup.md (hunk 10 above)
Comment (prune998) [reply docs/shell_setup.md]: install is a submenu of `Oh My ZSH !` so I guess it's ok ?

**Review by archyufa — COMMENTED** [review]
few more comments

File: docs/shell_setup.md (hunk 13 above)
Comment (prune998) [reply docs/shell_setup.md:159]: kube-ps1 is already taken care by powerLevel10k

looking into the others, but I guess they are too...

File: docs/shell_setup.md (hunk 14 above)
Comment (prune998) [reply docs/shell_setup.md:163]: I added a section, will resolve the issue for now

**Pull request merged**